roles/openshift_health_checker/openshift_checks/package_version.py

```diff
@@ -17,9 +30,86 @@ def is_active(cls, task_vars):
         return super(PackageVersion, cls).is_active(task_vars) and master_or_node
 
     def run(self, tmp, task_vars):
+        rpm_prefix = get_var(task_vars, "openshift", "common", "service_type")
+        openshift_release = get_var(task_vars, "openshift_release", default='')
+        deployment_type = get_var(task_vars, "openshift_deployment_type")
+        check_multi_minor_release = deployment_type in ['openshift-enterprise']
+
         args = {
-            "requested_openshift_release": get_var(task_vars, "openshift_release", default=''),
-            "openshift_deployment_type": get_var(task_vars, "openshift_deployment_type"),
-            "rpm_prefix": get_var(task_vars, "openshift", "common", "service_type"),
+            "package_list": [
+                (
+                    "openvswitch",
+                    self.get_required_ovs_version(task_vars),
+                    False
+                ),
+                (
+                    "docker",
+                    self.get_required_docker_version(task_vars),
+                    False
+                ),
+                (
+                    "{}".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+                (
+                    "{}-master".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+                (
+                    "{}-node".format(rpm_prefix),
+                    openshift_release,
+                    check_multi_minor_release
+                ),
+            ],
         }
+
         return self.execute_module("aos_version", args, tmp, task_vars)
+
+    def get_required_ovs_version(self, task_vars):
+        """Return the correct Open vSwitch version for the current OpenShift version.
+        If the current OpenShift version is >= 3.5, ensure Open vSwitch version 2.6,
+        Else ensure Open vSwitch version 2.4"""
+        openshift_version = self.get_openshift_version(task_vars)
+
+        if openshift_version < 3.5:
+            return self.openshift_to_ovs_version["3.4"]
+
+        if not self.openshift_to_ovs_version.get(str(openshift_version)):
```

Comment:
nit but I'd like the logic better as:

        ovs_version = self.openshift_to_ovs_version.get(str(openshift_version))
        if ovs_version:
            return ovs_version

        msg = "There is no recommended version of Open vSwitch for the current version of OpenShift: {}"
        raise OpenShiftCheckException(msg.format(openshift_version))